include/xgboost/predictor.h

```diff
@@ -0,0 +1,172 @@
+/*!
+ * Copyright by Contributors
+ * \file predictor.h
+ * \brief Interface of predictor,
+ *  performs predictions for a gradient booster.
+ */
+#pragma once
+#include <xgboost/base.h>
+#include <functional>
+#include <memory>
+#include <vector>
+#include <string>
+#include "../../src/gbm/gbtree_model.h"
+
+// Forward declarations
+namespace xgboost {
+class DMatrix;
+class TreeUpdater;
+}
+namespace xgboost {
+namespace gbm {
+struct GBTreeModel;
+}
+}  // namespace xgboost
+
+namespace xgboost {
+
+/**
+ * \class Predictor
+ *
+ * \brief Performs prediction on individual training instances or batches of instances for GBTree.
+ *        The predictor also manages a prediction cache associated with input matrices. If possible,
+ *        it will use previously calculated predictions instead of calculating new predictions.
+ *        Prediction functions all take a GBTreeModel and a DMatrix as input and output a vector of
+ *        predictions. The predictor does not modify any state of the model itself.
+ */
+
+class Predictor {
+ public:
+  virtual ~Predictor() {}
+
+  /**
+   * \fn  void Predictor::InitCache(const std::vector<std::shared_ptr<DMatrix> > &cache);
```

Comment:
This was automatically generated. I will avoid doing this in future in that case.